api/v1beta1/agentserviceconfig_types.go

```diff
@@ -122,6 +122,31 @@ type AgentServiceConfigSpec struct {
 	// +optional
 	OSImageAdditionalParamsRef *corev1.LocalObjectReference `json:"OSImageAdditionalParamsRef,omitempty"`
 
+	// Ingress contains configuration for the ingress resources.
+	// Has no effect when running on an OpenShift cluster.
+	//+operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Ingress"
+	// +optional
+	Ingress Ingress `json:"ingress,omitempty"`
+}
+
+type Ingress struct {
+	// AssistedServiceHostname is the hostname to be assigned to the assisted-service ingress.
+	// Has no effect when running on an OpenShift cluster.
+	//+operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Assisted Service hostname"
+	// +optional
+	AssistedServiceHostname string `json:"assistedServiceHostname,omitempty"`
```

Comment:
Are these really optional? What happens when they're not provided? I see that the string gets plumbed through to the Ingress, so if it's `""`, is that ok?

Reply:
Ah I missed removing that when nesting it. Thanks for catching this.